docs/core/whats-new/dotnet-7.md

```diff
@@ -0,0 +1,179 @@
+
+For more information, see [Central package management](/nuget/consume-packages/Central-Package-Management).
+
+## P/Invoke source generation
+
+.NET 7 introduces a source generator for platform invokes (P/Invokes) in C#. The source generator looks for <xref:System.Runtime.InteropServices.LibraryImportAttribute> on `static`, `partial` methods to trigger compile-time source generation of marshalling code. By generating the marshalling code at compile time, no IL stub needs to be generated at run time, as it does when using <xref:System.Runtime.InteropServices.DllImportAttribute>. The source generator improves application performance and also allows the app to be ahead-of-time (AOT) compiled. For more information, see [Source generation for platform invokes](../../standard/native-interop/pinvoke-source-generation.md) and [Use custom marshallers in source-generated P/Invokes](../../standard/native-interop/tutorial-custom-marshaller.md).
+
+## Related releases
+
+This section contains information about related products that have releases to coincide with the .NET 7 release.
+
+<!--
+### Visual Studio 2022 version 17.4
+
+For more information, see [What's new in Visual Studio 2022](/visualstudio/ide/whats-new-visual-studio-2022).
+-->
+
+### C# 11
+
+C# 11 includes support for [generic math](../../standard/generics/math.md), raw string literals, file-scoped types, and other new features. For more information, see [What's new in C# 11](../../csharp/whats-new/csharp-11.md).
+
+### .NET MAUI
+
+.NET Multi-platform App UI (.NET MAUI) is a cross-platform framework for creating native mobile and desktop apps with C# and XAML. It unifies Android, iOS, macOS, and Windows APIs into a single API. For information about the latest updates, see [...](/dotnet/maui/...).
+
+### ASP.NET Core
+
+ASP.NET Core 7.0 includes rate-limiting middleware, improvements to minimal APIs, and gRPC JSON transcoding. For information about all the updates, see [What's new in ASP.NET Core 7](/aspnet/core/release-notes/aspnetcore-7.0).
+
+### EF Core
+
+Entity Framework Core 7.0 includes provider-agnostic support for JSON columns, improved performance for saving changes, and custom reverse engineering templates. For information about all the updates, see [What's new in EF Core 7.0](/ef/core/what-is-new/ef-core-7.0/whatsnew).
+
+### Windows Forms
+
+Much work has gone into Windows Forms for .NET 7. Improvements have been made in the following areas:
+
+- Accessibility
+- High DPI and scaling
+- Databinding
+
+For more information, see [What's new in Windows Forms in .NET 7](https://devblogs.microsoft.com/dotnet/winforms-enhancements-in-dotnet-7).
+
+<!--
+### Orleans
```

Comment:
I think you can uncomment this out, as long as we plan to release these at the same time. My Orleans article will be ready by then.

Reply:
Do you want to target the `release-dotnet-7` branch for that PR?